tests/ConnectionTest.php

```diff
@@ -37,11 +37,101 @@ public function testDb()
+            'config' => [
+                'host' => 'some-host',
+                'database' => 'tests',
+            ],
+        ];
+
+        yield 'Host and port' => [
+            'expectedUri' => 'mongodb://some-host:12345',
+            'expectedDatabaseName' => 'tests',
+            'config' => [
+                'host' => 'some-host',
+                'port' => 12345,
+                'database' => 'tests',
+            ],
+        ];
+
+        yield 'Port in host name takes precedence' => [
+            'expectedUri' => 'mongodb://some-host:12345',
+            'expectedDatabaseName' => 'tests',
+            'config' => [
+                'host' => 'some-host:12345',
+                'port' => 54321,
+                'database' => 'tests',
+            ],
+        ];
+
+        yield 'Multiple hosts' => [
+            'expectedUri' => 'mongodb://host-1,host-2',
+            'expectedDatabaseName' => 'tests',
+            'config' => [
+                'host' => ['host-1', 'host-2'],
+                'database' => 'tests',
+            ],
+        ];
+
+        yield 'Multiple hosts with same port' => [
+            'expectedUri' => 'mongodb://host-1:12345,host-2:12345',
+            'expectedDatabaseName' => 'tests',
+            'config' => [
+                'host' => ['host-1', 'host-2'],
+                'port' => 12345,
+                'database' => 'tests',
+            ],
+        ];
+
+        yield 'Multiple hosts with port' => [
+            'expectedUri' => 'mongodb://host-1:12345,host-2:54321',
+            'expectedDatabaseName' => 'tests',
+            'config' => [
+                'host' => ['host-1:12345', 'host-2:54321'],
+                'database' => 'tests',
+            ],
+        ];
+
+        yield 'DSN takes precedence over host/port config' => [
+            'expectedUri' => 'mongodb://some-host:12345/auth-database',
+            'expectedDatabaseName' => 'tests',
+            'config' => [
+                'dsn' => 'mongodb://some-host:12345/auth-database',
+                'host' => 'wrong-host',
+                'port' => 54321,
+                'database' => 'tests',
+            ],
+        ];
+
+        yield 'Database is extracted from DSN if not specified' => [
+            'expectedUri' => 'mongodb://some-host:12345/tests',
+            'expectedDatabaseName' => 'tests',
+            'config' => [
+                'dsn' => 'mongodb://some-host:12345/tests',
+            ],
+        ];
+    }
+
+    /** @dataProvider dataConnectionConfig */
+    public function testConnectionConfig(string $expectedUri, string $expectedDatabaseName, array $config): void
+    {
+        $connection = new Connection($config);
+        $client = $connection->getMongoClient();
+
+        $this->assertSame($expectedUri, (string) $client);
+        $this->assertSame($expectedDatabaseName, $connection->getMongoDB()->getDatabaseName());
+    }
+
+    public function testConnectionWithoutConfiguredDatabase(): void
```

Comment:
This likely answers my previous question about the purpose of `database` in the config struct.